thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/AnomalyResource.java

```diff
@@ -563,26 +594,175 @@ public Response applyAutotuneConfig(@PathParam("autotuneConfigId") @NotNull long
-          startTime, endTime, dimensionMap);
-      dimensionMapAnomalyTimelinesViewMap.put(dimensionMap.toString(), anomalyTimelinesView);
+    for(DimensionMap dimensionMap : dimensions) {
+      AnomalyTimelinesView anomalyTimelinesView = null;
+      if(mode.equalsIgnoreCase("ONLINE")) {
+        // If online, use the monitoring time window to query time range and generate the baseline
+        anomalyTimelinesView = anomaliesResource.getTimelinesViewInMonitoringWindow(anomalyFunctionSpec,
+            datasetConfigDTO, startTime, endTime, dimensionMap);
+      } else {
+        // If offline, request baseline in user-defined data range
+        List<Pair<Long, Long>> dataRangeIntervals = new ArrayList<>();
+        dataRangeIntervals.add(new Pair<Long, Long>(endTime.getMillis(), endTime.getMillis()));
+        dataRangeIntervals.add(new Pair<Long, Long>(startTime.getMillis(), endTime.getMillis()));
+        anomalyTimelinesView = anomaliesResource.getTimelinesViewInMonitoringWindow(anomalyFunctionSpec,
+            datasetConfigDTO, dataRangeIntervals, dimensionMap);
+      }
+      dimensionMapAnomalyTimelinesViewMap.put(dimensionMap.toJavaString(), anomalyTimelinesView);
+    }
+
+    // Amend the anomaly time lines view by adding merged anomaly information
+    for (String dimension : dimensionMapAnomalyTimelinesViewMap.keySet()) {
+      DimensionMap dimensionMap = new DimensionMap();
+      AnomalyTimelinesView anomalyTimelinesView = amendAnomalyTimelinesViewWithAnomalyResults(anomalyFunctionSpec,
+          dimensionMapAnomalyTimelinesViewMap.get(dimension), dimensionMap);
+      dimensionMapAnomalyTimelinesViewMap.put(dimension, anomalyTimelinesView);
+    }
+
+    return Response.ok(dimensionMapAnomalyTimelinesViewMap).build();
+  }
+
+  /**
+   * Amend the anomaly time lines view with the information in anomaly results
+   *  - if the anomaly result contains view, override the timeseries by the view
+   *  - if the anomaly result contains raw anomalies, override the timeseries by the current and baseline values inside
+   *  - otherwise, use the merged anomaly information for all missing points
+   * @param anomalyFunctionSpec
+   * @param originalTimelinesView
+   * @param dimensionMap
+   * @return
+   */
+  public AnomalyTimelinesView amendAnomalyTimelinesViewWithAnomalyResults(AnomalyFunctionDTO anomalyFunctionSpec,
+      AnomalyTimelinesView originalTimelinesView, DimensionMap dimensionMap) {
+    if (originalTimelinesView == null || anomalyFunctionSpec == null) {
+      LOG.error("AnomalyTimelinesView or AnomalyFunctionDTO is null");
+      return null;
+    }
+    long bucketMillis =
+        (new TimeGranularity(anomalyFunctionSpec.getBucketSize(), anomalyFunctionSpec.getBucketUnit())).toMillis();
+    Tuple2<TimeSeries, TimeSeries> timeSeriesTuple = TimeSeriesUtils.toTimeSeries(originalTimelinesView);
```

Comment:
We discussed about using Pair offline. However, Pair takes type implements Serializable; however, TimeSeries is not Serializable class. We are not able to use Pair.